modules/rest_api/controllers/services/rest.php

```diff
@@ -1832,19 +1854,88 @@ private function esGetSpecialFieldBlank(array $doc) {
    * @return string
    *   Combined comment string.
    */
-  private function esGetSpecialFieldSampleOccurrenceComment(array $doc) {
+  private function esGetSpecialFieldSampleOccurrenceComment(array $doc, array $params) {
     $oComment = isset($doc['occurrence']['occurrence_remarks']) ? $doc['occurrence']['occurrence_remarks'] : '';
     $sComment = isset($doc['event']['event_remarks']) ? $doc['event']['event_remarks'] : '';
+    if (!empty($params) && in_array("notab", $params)) {
+      $oComment = str_replace("\t", ' ', $oComment);
+      $sComment = str_replace("\t", ' ', $sComment);
+    }
+    if (!empty($params) && in_array("nonewline", $params)) {
+      $oComment = str_replace(array("\r\n", "\n", "\r"), ' ', $oComment);
+      $sComment = str_replace(array("\r\n", "\n", "\r"), ' ', $sComment);
+    }
     if ($oComment !== '' && $sComment !== '') {
-      return "Record comment: $oComment Sample comment: $sComment";
+      $comment = "$oComment $sComment";
     }
     elseif ($oComment !== '') {
-      return "Record comment: $oComment";
+      $comment =  $oComment;
     }
     elseif ($sComment !== '') {
-      return "Sample comment: $sComment";
+      $comment = $sComment;
     }
     else {
+      $comment = '';
+    }
+    if (!empty($params) && in_array("addref", $params)) {
+      // Can't access '_id' here so, for now, hard-coding
+      // appending 'brc1|' to ref. This will need to be addressed
+      // if we add another warehouse. 
+      $ref = $doc['id'];
+      $comment = trim("$comment brc1|$ref");
```

Comment:
Maybe the prefix could be a configuration in application/config/indicia.php?

Reply:
I just had an email from Martin - I definitely need to change this because the prefix on the live warehouse is iBRC - not brc1| - shall I go ahead and use a config option?

Reply:
@johnvanbreda - I've changed this to use a config parameter set in indicia_dist.php - `$config['es_key_prefix'] = 'iBRC';` - but I couldn't get rest.php to pick it up (though it can pick up `$config['warehouse_title']` okay). Is that some sort of caching problem?

Reply:
indicia_dist.php is the template that gets copied to indicia.php during install. I would suggest adding a commented out setting to indicia_dist.php, with an explanation comment that suggests you uncomment this if you want IDs to be prefixed in report downloads. Then for your development copy, indicia.php needs to have the setting. Can you please test that it works OK without the setting, i.e. defaults to an empty string, so existing warehouses don't break.

Reply:
That was silly of me. I've added to the template in the pull request - commented out and with explanation comment as you suggested. Got it to work on dev after putting in the actual config file. I'd already tested that it works when not set since I didn't have it set in the right file. Looks like kohana::config returns a string if the array element isn't found.

Reply:
@burkmar - OK, looks good. Do you want this merged now?

Reply:
Yes please @johnvanbreda - @kitenetter is happy with it so it is good to go. Will you let me know when this gets onto the live iRecord site so I can set up the form?

Reply:
@burkmarr everything has now been deployed (Warehouse and iRecord).

Reply:
Thanks @johnvanbreda - I'll get on with the form.